pycbc/inject/inject.py

```diff
@@ -46,13 +44,13 @@
 from pycbc.io.ligolw import LIGOLWContentHandler
 from ligo.lw import utils as ligolw_utils, ligolw, lsctables
 
+sim = libutils.import_optional('lalsimulation')
 
 injection_func_map = {
-    np.dtype(float32): sim.SimAddInjectionREAL4TimeSeries,
-    np.dtype(float64): sim.SimAddInjectionREAL8TimeSeries
+    np.dtype(float32): lambda *args: sim.SimAddInjectionREAL4TimeSeries(*args),
```

Comment:
I'm surprised that this wouldn't fail if lalsimulation is not present??

Reply:
It works because the lambda isn't actually evaluating the expression, whereas it would be if directly putting a function into a dictionary. It's a way to hide the use until it is used rather than when creating the dict.